In [9]:
import pandas as pd
import json
import geopandas as gpd

# Offstreet Parking in Sydney

In [2]:
data = pd.read_csv("../Datasets/OffstreetparkingData.csv", encoding = "ISO-8859-1") # Weird issue with non-ASCII char

In [3]:
data.columns

Index(['FRD_ID', 'Owner', 'Building_name_location', 'Street_Number_GPS',
       'Street_Name_GPS', 'Suburb_GPS', 'State', 'Postcode', 'Height_Res',
       'Total_number_of_bays', 'Access_Lat', 'Access_Long', 'Kerbside_Ref',
       'Between', 'Open', 'Close', 'Preferred', 'URL', 'Phone', 'Comments',
       'Road Direction Access', 'Kerb', 'Last updated'],
      dtype='object')

In [4]:
data

FRD_ID               Owner           Building_name_location  \
0   FRD-001  Sir Stamford Hotel              93 Macquarie Street   
1   FRD-002              Wilson                Gold Fields House   
2   FRD-003              Wilson       No. 1 OConnell St Car Park   
3   FRD-004              Wilson                The Chifley Tower   
4   FRD-005              Secure          Sofitel Wentworth Hotel   
5   FRD-006              Wilson                 Australia Square   
6   FRD-007      Capitol Square                   Capitol Square   
7   FRD-008              Secure                       Piccadilly   
8   FRD-009            Sheraton                         Sheraton   
9   FRD-010              Enacom                 Cathedral Street   
10  FRD-011              Secure                 The Bowlers Club   
11  FRD-012              Wilson                 St Martins Tower   
12  FRD-013              Secure                   1 Dixon Street   
13  FRD-014              Secure                234 Sussex Street   
14  FRD-015              Secure                  60 Elizabeth St   
15  FRD-016              Wilson                  Grosvenor Place   
16  FRD-017              Wilson                155 George Street   
17  FRD-018              Wilson      Overseas Passenger Terminal   
18  FRD-019              Secure  Goulburn Street Parking Station   
19  FRD-020              Secure  World Square (Goulbourn Street)   
20  FRD-021              Wilson                  Darling Quarter   
21  FRD-022              Wilson        121-127 Harrington Street   
22  FRD-023              Wilson              Clocktower Car Park   
23  FRD-024              Wilson                        Quay West   
24  FRD-025           InterPark                   Bond Store One   
25  FRD-026              Wilson        Barangaroo Point car park   
26  FRD-027              Secure                      Amora Hotel   
27  FRD-028              Secure                       Met Centre   
28  FRD-029       Cinema Centre                  521 Kent Street   
29  FRD-030              Secure                  528 Kent Street   
..      ...                 ...                              ...   
48  FRD-050              Wilson           Governor Phillip Tower   
49  FRD-051              Secure             131 Macquarie Street   
50  FRD-052              Secure                  109 Pitt Street   
51  FRD-053              Secure                           Atrium   
52  FRD-054              Secure                           Hilton   
53  FRD-055              Secure                No 1 Martin Place   
54  FRD-056              Wilson                   37 Pitt Street   
55  FRD-057              Wilson             Angel Place Car Park   
56  FRD-058              Wilson                 Citigroup Centre   
57  FRD-059              Wilson        Market City - Level B2/B3   
58  FRD-060              Wilson           Market City - Level B1   
59  FRD-061              Wilson                          Gateway   
60  FRD-062              Secure                King Street Wharf   
61  FRD-063           InterPark                           Domain   
62  FRD-064              Secure                132 Sussex Street   
63  FRD-065       Sussex Centre                401 Sussex Street   
64  FRD-066              Wilson                  321 Kent Street   
65  FRD-067              Secure                   Allianz Centre   
66  FRD-068              Wilson                     Darling Park   
67  FRD-069              Wilson                          Novatel   
68  FRD-070              Wilson                187 Thomas Street   
69  FRD-071              Wilson                      HSBC Centre   
70  FRD-073  Wynyard Properties                     Wynyard Lane   
71  FRD-074              Wilson                   71 York Street   
72  FRD-075              Secure                         Landmark   
73  FRD-076              Wilson          Queen Victoria Building   
74  FRD-077              Wilson                 201 Elizabeth

We got the number of bays for parking. Note that they are all in the 2000 postcode.

In [5]:
data.Total_number_of_bays.sum()

21417.0

There are around 21417 available car parks in the Sydney. 

# Paid Street Parking in Sydney

In [71]:
#load json object
with open('../Datasets/paid_street_parking.json') as f:
    d = json.load(f)
    f.close()

In [72]:
import csv
def initial_write():
    '''Create and write first entry in file.'''
    csv_file = "paid_street_parking" + '.csv'
    write_file = open(csv_file, 'w')
    writer = csv.writer(write_file)
    list_of_keys = []
    for x in d[0]['attributes'].keys():
        list_of_keys.append(x)
    for x in d[0]['geometry'].keys():
        list_of_keys.append(x)
    writer.writerow(list_of_keys)
    write_file.close()

In [73]:
initial_write()

In [74]:
def write_to_csv():
    '''Writes current json entry into csv'''
    csv_file = "paid_street_parking" + '.csv'
    write_file = open(csv_file, 'a')
    writer = csv.writer(write_file)
    list_of_values = []
    for i in range(0, len(d)):
        list_of_values.clear()
        for x in d[i]['attributes'].values():
            list_of_values.append(x)
        for x in d[i]['geometry'].values():
            list_of_values.append(x)
        writer.writerow(list_of_values)
    write_file.close()

In [75]:
write_to_csv()

In [76]:
data_two = pd.read_csv('../Datasets/paid_street_parking.csv')

In [78]:
sum(data_two.ApproxPaySpaces)

2460

2460 parking spaces across Sydney.

In [84]:
data_two.groupby("Suburb").sum().ApproxPaySpaces

Suburb
Camperdown        25
Chippendale       80
Darlinghurst     454
Dawes Point      195
Elizabeth Bay      1
Glebe            118
Green Square      20
Haymarket         45
Millers Point    265
Newtown           54
Potts Point       22
Pyrmont          177
Surry Hills      288
Sydney           481
The Rocks         85
Ultimo           150
Name: ApproxPaySpaces, dtype: int64

ApproxpaySpaces tells us the number of available street parking spots in that suburb.